config_utilities/include/config_utilities/config.h

```diff
@@ -255,4 +257,42 @@ inline void check(Ts... args) {
   internal::Visitor::visitCheck(CheckType(std::forward<Ts>(args)...));
 }
 
+/**
+ * @brief Update a field in a config.
+ * @note This function will update the field and check the validity of the config afterwards. If the config is invalid,
+ * the field will be reset to its original value.
+ * @param node The node containing the field to update.
+ */
+template <typename ConfigT>
+bool updateField(ConfigT& config, YAML::Node node, const std::string& field_name, const std::string& name_space = "") {
+  const auto old_value = internal::Visitor::getValues(config, true, name_space, field_name);
+  internal::Visitor::setValues(config, internal::lookupNamespace(node, name_space));
+  bool success = isValid(config);
+
+  if (!success) {
+    internal::Visitor::setValues(config, internal::lookupNamespace(old_value.data, name_space));
+  }
+
+  return success;
```

Comment:
```suggestion
  const MetaData data=  internal::Visitor::setValues(config, internal::lookupNamespace(node, name_space));
  return !data.hasErrors();
```

That should do exactly what the initial code is intending to do. Only thing to recall is that this does not check for the specific field name and will set all params whose name is in the node. If you want that behavior the yaml node needs to be filtered. Other thoughts:
- We could add a bool to print or not print warnings that result from setting the values (e.g. conversion failures etc)
- Could also add these functionalities [here](https://github.com/MIT-SPARK/config_utilities/blob/main/config_utilities/include/config_utilities/parsing/yaml.h#L62) (and also in the ROS parsing, then all configs can be updated from YAML and ROS)